### ISSUE TITLE
no-shadow: false positive when importing type

### ISSUE THREAD

**Issue by OliverJAsh** [typescript-eslint/typescript-eslint#550]
**What code were you trying to parse?**

This works fine, as expected.

```ts
type Consumer = {};
const fn = (Consumer: Consumer) => {};
```

However, if the type is imported, this errors unexpectedly:

```ts
import { Consumer } from './other';

// ESLint error: 'Consumer' is already declared in the upper scope.
const fn = (Consumer: Consumer) => {};
```

**What did you expect to happen?**
No error, because there is no shadowing. Here I am defining a parameter that has the same name as the type, but this doesn't count as shadowing, because one is a type, one is a value.

**What actually happened?**
Unexpected error.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.90` |
| `TypeScript`                | `3.4.1` |
| `ESLint`                    | `5.16.0` |
| `node`                      | `8.15.1` |
| `npm`                       | `6.4.1` |

**Comment by mcmar** [comment]
I'm not sure if this is related to this issue or not. The `no-shadow` rule also is triggered by typescript module augmentation:
```
import { pathUtils } from 'react-navigation';

declare module 'react-navigation' {
  // this sets off no-shadow rule
  const pathUtils: {
    urlToPathAndParams(url: string, uriPrefix: string): { path: string; params: {} };
  };
}
```

**Comment by frenic** [comment]
This also happens on enums:

```ts
import { Consumer } from './other';

enum UserType {
  Customer, // ESLint error: 'Customer' is already declared in the upper scope
}
```

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.7.0` |
| `TypeScript`                | `3.7.2` |
| `ESLint`                    | `6.6.0` |
| `node`                      | `12.4.0` |
| `yarn`                       | `1.15.2` |

**Comment by bradzacher** [comment]
Merging into #1856
